packages/scandipwa/src/component/CheckoutOrderSummary/CheckoutOrderSummary.component.js

```diff
@@ -117,6 +123,47 @@ export class CheckoutOrderSummary extends PureComponent {
         );
     }
 
+    renderDiscountCode() {
+        const {
+            totals: { coupon_code, items },
+            checkoutStep,
+            isMobile
+        } = this.props;
+
+        if (!items || items.length < 1) {
+            return null;
+        }
+
+        if (isMobile && checkoutStep === BILLING_STEP) {
+            return (
+                <div>
+                    <div
+                      block="ExpandableContent"
+                      elem="Heading"
+                      mix={ { block: 'CartPage', elem: 'ExpandableContentHeading' } }
+                    >
+                    Have a discount code?
+                    </div>
+                    <CartCoupon couponCode={ coupon_code } />
+                </div>
+            );
+        }
+
+        if (checkoutStep === BILLING_STEP) {
+            return (
+                <ExpandableContent
+                  heading={ __('Have a discount code?') }
+                  mix={ { block: 'CartPage', elem: 'Discount' } }
+                  isArrow
+                >
+                    <CartCoupon couponCode={ coupon_code } />
+                </ExpandableContent>
+            );
+        }
+
+        return null;
```

Comment:
Make it simplier, can even create separate function for rendering mobile version.

1. checkout step is not billing step => return null;
2. if is mobile => render mobile version;
3. default render just expandablecontent.